src/pages/LandingPage/WishListForm.jsx

```diff
@@ -0,0 +1,154 @@
+import { yupResolver } from '@hookform/resolvers/yup'
+import { styled } from '@mui/material'
+import React, { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { UploadImage } from '../../components/UploadImage'
+import { wishListSchema } from '../../utils/helpers/validate'
+import { SelectComponent } from '../../components/UI/SelectComponent'
+import { TextArea } from '../../components/UI/TextArea'
+import { options } from '../../utils/constants/options'
+import { DatePicker } from '../../components/DatePicker'
+import { Button } from '../../components/UI/Button'
+import { Input } from '../../components/UI/input/Input'
+
+export const WishListForm = ({ onClose }) => {
+   const [error, setError] = useState(null)
+   const [preview, setPreview] = useState({ file: '' })
+   const {
+      register,
+      handleSubmit,
+      formState: { errors },
+      reset,
+      control,
+   } = useForm({
+      mode: 'onBlur',
+      resolver: yupResolver(wishListSchema),
+   })
+
+   const onSubmit = (data) => {
+      console.log(data)
+      reset()
+      setPreview(null)
+   }
+
+   const onError = (error) => {
+      if (error === 'invalidDate') {
+         setError('Неверный формат даты')
+      }
+      if (error === '') {
+         setError('Укажите дату')
+      }
+   }
+
+   return (
+      <Container>
+         <BlockOne>
+            <UploadImage preview={preview} setPreview={setPreview} />
+         </BlockOne>
+         <BlockTwo onSubmit={handleSubmit((data) => onSubmit(data, preview))}>
+            <h3>Добавление желаемого подарка</h3>
+
+            <InputContainer>
+               <StyledInput
+                  labelText="Название подарка"
+                  placeholder="Введите название подарка"
+                  {...register('holidayName')}
+                  error={Boolean(errors.holidayName)}
+                  helperText={errors.holidayName?.message}
+               />
+
+               <StyledInput
+                  labelText="Ссылка на подарок"
+                  placeholder="Вставьте ссылку на подарок"
+                  {...register('link')}
+                  error={Boolean(errors.link)}
+                  helperText={errors.link?.message}
+               />
+
+               <SelectComponent
+                  data={options}
+                  label="Праздник"
+                  isButton="true"
+                  placeholder="Выберите праздник "
+                  name="holiday"
+                  control={control}
+                  error={Boolean(errors.holiday)}
+                  helperText={errors.holiday?.message}
+               />
+               <StyledDatePicker
```

Comment:
[Screencast from 2023-11-02 13-01-56.webm](https://github.com/peaksoft-school/gift-list-js10/assets/110243235/5df5ff56-d9fa-460c-8f53-7a9252092712)

Reply:
and please write name in English.